src/plugins/share/public/types.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  EuiContextMenuPanelDescriptor,
+  EuiContextMenuPanelItemDescriptor,
+} from '@elastic/eui/src/components/context_menu/context_menu';
+
+/**
+ * @public
+ * Properties of the current object to share. Registered share
+ * menu providers will provide suitable items which have to
+ * be rendered in an appropriate place by the caller.
+ *
+ * It is possible to use the static function `showShareContextMenu`
+ * to render the menu as a popover.
+ * */
+export interface ShareMenuItemProps {
+  objectType: string;
+  objectId?: string;
+  /**
+   * Current url for sharing. This can be set in cases where `window.location.href`
+   * does not contain a shareable URL (e.g. if using session storage to store the current
+   * app state is enabled). In these cases the property should contain the URL in a
+   * format which makes it possible to use it without having access to any other state
+   * like the current session.
+   *
+   * If not set it will default to `window.location.href`
+   */
+  shareableUrl: string;
+  sharingData: unknown;
+  isDirty: boolean;
```

Comment:
It's also passed into the `getShareMenuItems` call of the registered providers and they act on it: `x-pack/legacy/plugins/reporting/public/share_context_menu/register_csv_reporting.tsx`

But you are right, we don't have to pass it into the component anymore because the registered providers are now queried outside of the component. I will change that.

Reply:
Did this get changed yet? Not a huge deal if it isn't possible to remove, but if it is, I'd love to see it go 💥

Reply:
No, in this place it's still used, but it's gone in the component itself: `src/plugins/share/public/components/share_context_menu.tsx`